src/Platform/PlatformWindow.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <string>
+#include <memory>
+
+#include "Utility/Geometry/Size.h"
+#include "Utility/Geometry/Point.h"
+
+#include "PlatformOpenGLOptions.h"
+
+class PlatformOpenGLContext;
+
+/**
+ * Abstraction for accessing platform-specific window API.
+ *
+ * This class is essentially an API handle and doesn't hold any state. Windows users can think about it as
+ * a thin wrapper atop of `HWND`, so a name like `PlatformWindowApiHandle` might've been more suitable here.
+ * Fortunately we're not writing in Java and prefer shorter names.
+ *
+ * Event processing & state storage can be done by subclassing `PlatformEventHandler` and then using it when calling
+ * into `PlatformEventLoop`.
+ */
+class PlatformWindow {
+ public:
+    virtual ~PlatformWindow() = default;
+
+    virtual void SetTitle(const std::string &title) = 0;
+    virtual std::string Title() const = 0;
+
+    virtual void Resize(const Sizei &size) = 0;
+    virtual Sizei Size() const = 0;
+
+    virtual void SetPosition(const Pointi &pos) = 0;
+    virtual Pointi Position() const = 0;
+
+    virtual void SetVisible(bool visible) = 0;
+    virtual bool IsVisible() const = 0;
+
+    virtual void SetFullscreen(bool fullscreen) = 0;
+    virtual bool IsFullscreen() const = 0;
+
+    virtual void SetBorderless(bool borderless) = 0;
+    virtual bool IsBorderless() const = 0;
+
+    virtual void SetGrabsMouse(bool grabsMouse) = 0;
+    virtual bool GrabsMouse() const = 0;
```

Comment:
the naming is a bit inconsistent e.g SetFullscreen / IsFullscreen are using verbs (which i find nice) but then GrabsMouse is standing apart a little bit (IsMouseLocking?) and finally Size/SystemHandle (vs GetWidth)  is completely odd one out
let's pick 1 way and stick to it

Reply:
It's actually consistent. The rules are as follows:

1. Setters start with `Set`.
2. Getters are named the same as setters, just without the `Set`.
3. Boolean getters are prefixed with `Is` if the getter itself is an adjective. `Is` is not added if getter is a verb followed by an object ("object" here is object from the english grammar, a noun that's acted upon, not object from C++).
4. Mutators that are not setters are verbs.

So what we get is code that reads as grammatically correct English:
```
if(window->IsBorderless()) // if window is borderless
if(window->GrabsMouse()) // if window grabs mouse
if(window->Title() == "Blablabla") // If window title equals blablabla
```

Note that with `GrabsMouse`, an alternative name would be `IsMouseGrabber`. Both are OK, but I prefer `GrabsMouse` because it's shorter.

The only method that doesn't fit is `Resize`, I'm breaking my own rules here and following Qt naming.

GetWidth/GetHeight are to be removed later. I've left them for now because if I were to remove them then the diff would span 200+ files and not 150.

Reply:
Also, while you're here.

We need you in discord! Need to move the repo into an org & create a separate test data repository there, and only you can do this.